src/log.c

```diff
@@ -0,0 +1,361 @@
+{
+    uint64_t now;
+    
+    now = rte_get_timer_cycles();
+    
+    if (w_buf.pos && ((now - w_buf.time) >= timeout*rte_get_timer_hz())) {
+        log_buf_flush(f);
+    }
+    return 0;
+}
+
+static int log_slave_process(void)
+{
+    struct dpvs_log *msg_log;
+    int ret = EDPVS_OK;
+    FILE *f = rte_logs.file;
+
+    /* dequeue LOG from ring, no lock for ring and w_buf */
+    while (0 == rte_ring_dequeue(log_ring, (void **)&msg_log)) { 
+        if (msg_log->log_len > LOG_BUF_MAX_LEN) {
+            dpvs_log_free(msg_log);
+            continue;
+        }
+        if (w_buf.pos + msg_log->log_len >= LOG_BUF_MAX_LEN) {
+            log_buf_flush(f);
+        }
+        
+        if (!w_buf.pos) {
+            w_buf.level = msg_log->log_level - 1;
+            w_buf.time = rte_get_timer_cycles();        
+        }
+        strncpy(w_buf.buf+w_buf.pos, msg_log->data, msg_log->log_len);
+        w_buf.pos += msg_log->log_len;
+
+        log_buf_timeout_flush(f, 5);            
+        dpvs_log_free(msg_log);
+    }
+    
+    log_buf_timeout_flush(f, 5);
+    
+    return ret;
+}
+
+static void log_slave_loop_func(void)
+{
+    g_dpvs_log_thread_ready = 1;
+    while(1){
+        log_slave_process();
+    }
+}
+
+static void log_signal_handler(int signum)
+{
+    if (signum == SIGABRT || signum == SIGSEGV) {
+        printf("\nSignal %d received, preparing to exit...\n",
+                signum);
+    }
+    log_slave_process();
+    log_buf_flush(rte_logs.file); 
+    signal(signum, SIG_DFL);
+    kill(getpid(), signum);
+}
+
+int log_slave_init(void)
+{
+    char ring_name[16];
+    int lcore_id;
+    FILE *f = rte_logs.file;
+#if CONFIG_DPVS_LOG_POOL_DEBUG
+    char log_pool_name[32];
+#endif
+
+    if (f != NULL) {
+        g_dpvs_log_time_off = LOG_SYS_TIME_LEN;
+    }
+            
+    RTE_LCORE_FOREACH_SLAVE(lcore_id) {
+        if (rte_eal_get_lcore_state(lcore_id) == FINISHED) {
+            rte_eal_wait_lcore(lcore_id);
+            dpvs_log_thread_lcore_set(lcore_id);
+            break;
+        }
+    }
+    snprintf(ring_name, sizeof(ring_name), "log_ring_%d", g_dpvs_log_core);
+    log_ring = rte_ring_create(ring_name, DPVS_LOG_RING_SIZE_DEF,
+                   rte_socket_id(), 0/*RING_F_SC_DEQ*/);
+    if (unlikely(NULL == log_ring)) {
+        fprintf(stderr, "Fail to init log slave core\n");
+        return EDPVS_DPDKAPIFAIL;
+    }
+#if CONFIG_DPVS_LOG_POOL_DEBUG
+    /* use memory pool for log msg */
+    snprintf(log_pool_name, sizeof(log_pool_name), "log_msg_pool");
+    dp_vs_log_pool = rte_mempool_create(log_pool_name,
+                                log_pool_size,
+                                sizeof(struct dpvs_log) + DPVS_LOG_MAX_LINE_LEN,
+                                log_pool_cache,
+                                0, NULL, NULL, NULL, NULL,
+                                0, 0);
+    if (!dp_vs_log_pool) {
+        return EDPVS_DPDKAPIFAIL;
+    }
+#endif    
```

Comment:
1. Can we use memory pool in default? 
2. `CONFIG_DPVS_LOG_POOL_DEBUG` is a little confusing, `CONFIG_DPVS_LOG_MEM_POOL` is enough, i think.